dataset/src/test/scala/frameless/ops/MatchesAllTest.scala

```diff
@@ -0,0 +1,30 @@
+package frameless
+package ops
+
+import shapeless.test.illTyped
+import shapeless.{::, Generic, HNil}
+
+private case class XString(s1: String, s2: String, s3: String)
+private case class XLong(l1: Long, l2: Long, l3: Long)
+
+class MatchesAllTest extends TypedDatasetSuite {
```

Comment:
I would just make this `object MatchesAllTest { ... }` since there is nothing happening at run time